trie/exceptions.py

```diff
@@ -177,6 +195,78 @@ def node(self) -> HexaryTrieNode:
         """
         return self.args[1]
 
+    @property
+    def untraversed_tail(self) -> Nibbles:
+        """
+        The nibbles that only reached partially into the extension or leaf node.
+        """
+        return self.args[2]
+
+    @property
+    def simulated_node(self) -> HexaryTrieNode:
+        """
+        For the purposes of walking a trie, we might only be interested in the sub_segments,
+        suffix, etc, of the node -- but assuming we actually had a node immediately at the
+        requested prefix. This returns a node simulated as if that were true.
+
+        See the trie walk tests for an example of how this is used.
+        """
+        return self._simulated_node
+
+    def _make_simulated_node(self) -> HexaryTrieNode:
+        from trie.utils.nodes import (
```

Comment:
Any specific reason for the local import?

Reply:
Yeah, the `exceptions` module usually shouldn't import from the rest of the project, because it often will cause an import cycle. I *think* it actually does cycle here if I try to import at the top. Even if it doesn't though, I don't want to leave a landmine for the next contributor.